src/main/java/com/contrastsecurity/maven/plugin/AbstractContrastMojo.java

```diff
@@ -0,0 +1,196 @@
+package com.contrastsecurity.maven.plugin;
+
+import com.contrastsecurity.sdk.ContrastSDK;
+import java.net.Authenticator;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.settings.Settings;
+
+/**
+ * Abstract mojo for mojos that need to connect to Contrast. Handles authentication, organization
+ * selection (multi-tenancy), and proxy configuration.
+ *
+ * <p>Extensions of this class use the {@link #connectToContrast()} to obtain an instance of the
+ * {@link ContrastSDK} with which they can make requests to Contrast.
+ */
+abstract class AbstractContrastMojo extends AbstractMojo {
+
+  @Parameter(defaultValue = "${settings}", readonly = true)
+  private Settings settings;
+
+  /**
+   * User name for communicating with Contrast. Agent users lack permissions required by this
+   * plugin. <a href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your personal
+   * keys</a>
+   */
+  @Parameter(alias = "username", required = true)
+  private String userName;
+
+  /**
+   * API Key for communicating with Contrast. <a
+   * href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your personal keys</a>
+   */
+  @Parameter(property = "apiKey", required = true)
+  private String apiKey;
+
+  /**
+   * Service Key for communicating with Contrast. <a
+   * href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your personal keys</a>
+   */
+  @Parameter(property = "serviceKey", required = true)
+  private String serviceKey;
+
+  /** Contrast API URL */
+  @Parameter(alias = "apiUrl", defaultValue = "https://app.contrastsecurity.com/Contrast")
+  private String url;
+
+  /**
+   * Unique ID for the Contrast Organization to which the plugin reports results. <a
+   * href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your Organization ID</a>
+   */
+  // TODO[JG] must this be required? If a user is only in one org, we can look it up using the
+  // endpoint /ng/profile/organizations
+  @Parameter(alias = "orgUuid", required = true)
```

Comment:
`orgUuid` is inconsistent with how we would name this property if it were included in the common config spec (which I think we should do, but I need to have a larger discussion about integrations and common config before we start the Gradle work).

I renamed it `organizationId` which is more consistent with common config and makes the UUID piece an implementation detail. The alias allows existing users to continue to specify this as `orgUuid`